ezidapp/management/commands/proc-crossref.py

```diff
@@ -39,161 +49,190 @@ class Command(ezidapp.management.commands.proc_base.AsyncProcessingCommand):
     setting = 'DAEMONS_CROSSREF_ENABLED'
     queue = ezidapp.models.async_queue.CrossrefQueue
 
-    def create(self, task_model):
-        pass
+    def run(self):
+        """Run async processing loop forever.
 
-    def update(self, task_model):
-        pass
+        This method is not called for disabled async processes.
+        """
+        while True:
+            qs = self.queue.objects.filter(
+                Q(status=self.queue.UNSUBMITTED)
+                | Q(status=self.queue.UNCHECKED)
+                | Q(status=self.queue.SUBMITTED)
+            ).order_by("seq")[: django.conf.settings.DAEMONS_MAX_BATCH_SIZE]
 
-    def delete(self, task_model):
-        pass
+            if not qs:
+                self.sleep(django.conf.settings.DAEMONS_IDLE_SLEEP)
+                continue
 
-    def _notOne(self, n):
-        if n == 0:
-            return "no"
+            for task_model in qs:
+                try:
+                    self.do_task(task_model)
+                except ezidapp.management.commands.proc_base.AsyncProcessingIgnored:
+                    log.debug(f'Ignored: {task_model.refIdentifier.identifier}')
+                    task_model.status = self.queue.IGNORED
+                except Exception as e:
+                    log.error('#' * 100)
+                    log.exception(f'Exception when handling task "{task_model}"')
+                    task_model.error = str(e)
+                    # if self.is_permanent_error(e):
+                    if True:
+                        task_model.status = self.queue.FAILURE
+                        task_model.errorIsPermanent = True
+
+                task_model.save()
+
+            self.sleep(django.conf.settings.DAEMONS_BATCH_SLEEP)
+
+    def do_task(self, task_model):
+        if task_model.status == self.queue.UNSUBMITTED:
+            self.submit(task_model)
+        elif task_model.status == self.queue.UNCHECKED:
+            self.check_status(task_model)
         else:
-            return "more than one"
+            raise AssertionError('Unhandled case')
```

Comment:
Why AssertionError? Perhaps ValueError? In any case, the AssertionError is fine for now